curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/6352ea3a5b79e614ee2282fd.md

```diff
@@ -0,0 +1,162 @@
+---
+id: 6352ea3a5b79e614ee2282fd
+title: Step 28
+challengeType: 0
+dashedName: step-28
+---
+
+# --description--
+
+There are a few edge cases to account for when calculating the mode of a dataset. First, if every value appears the same number of times, there is no mode.
+
+To calculate this, you will use a `Set`. A `Set` is a data structure that only allows unique values. If you pass an array into the `Set` constructor, it will remove any duplicate values.
+
+Start by creating an `if` statement. In the condition, create a new `Set()` and pass the `Object.values()` of your `counts` object. If the `size` property of this new `Set` is equal to `1`, that tells you every value appears the same number of times. In this case, return `null` from your function.
```

Comment:
Since this is the first time `Set` is introduced, this may help clear up some possible confusion about how to use it:

```suggestion
Start by creating an `if` statement. In the condition, create a `Set` with `new Set()` and pass it the `Object.values()` of your `counts` object. If the `size` property of this `Set` is equal to `1`, that tells you every value appears the same number of times. In this case, return `null` from your function.
```